zerver/models.py

```diff
@@ -2096,6 +2100,12 @@ class Meta:
         unique_together = (("realm", "name"),)
 
 
+class UserPushNotificationIdentity(models.Model):
+    id: int = models.AutoField(auto_created=True, primary_key=True, verbose_name="ID")
+    user_profile: UserProfile = models.OneToOneField(UserProfile, on_delete=CASCADE, null=False)
+    uuid: UUID = models.UUIDField(default=uuid4, unique=True)
+
+
```

Comment:
Why are we doing this as a new model, and not a new column on UserProfile?

Reply:
That's a good question. I can see two possible reasons:
* We currently exclude this table when doing a data export/import by default.
* The main downside of adding a column to `UserProfile` is that we fetch the value every time we fetch a `UserProfile` object.

Reply:
Excluding based on table is a valid reason -- though we'll need to add logic to regenerate it, and I'm not sure that's going to less effort than null'ing a field it on export and re-filling a field on import.

I don't know that I buy the cost of making the UserProfile table wider is actually a strong reason.  We don't fetch those rows in bulk, and we already have a _lot_ of crud on them.  A UUID seems far more useful to have in the core table than, say, shipping the `notification_sound` column around on every user fetch.

Reply:
I think the main argument is that this UUID is exclusively accessed when sending/registering for push notifications; whereas `notification_sound` is necessary for e.g. `/register` to display the user's settings. 

The other is that UUIDs compress poorly, and we do compress objects in our caches, but it may not really matter.

It's true that for our primary code paths where we fetch users in bulk, we do things like this:
```
@cache_with_key(realm_user_dicts_cache_key, timeout=3600 * 24 * 7)                                       
def get_realm_user_dicts(realm_id: int) -> List[Dict[str, Any]]:                                         
    return UserProfile.objects.filter(                                                                   
        realm_id=realm_id,                                                                               
    ).values(*realm_user_dict_fields)                                                                    
```

which only fetch specific fields. Our main `UserProfile` object cache does include full objects, but they tend to be fetched only 1-5 at a time, so it may not really matter, other than that changing `UserProfile` as a high-throughput model is a slightly more complex database migration.

I'll add that we have two other high-entropy (aka hard to compress) fields in UserProfile; `password` and `api_key`; the latter we plan to move to a separate table.

Reply:
We settled on a UserProfile field here; marking as resolved.